Core/DomainModel/EntityCollection.cs

```diff
@@ -127,6 +127,28 @@ public Entity FindEntity(Guid guid)
             return entities[guid];
         }
 
+        public List<Entity> FindEntityByTag(string tag)
+        {
+            return FindEntityByTag(new List<string> { tag });
```

Comment:
to find entities by a tag list, the intersection of the list containing the tags to find and the list of the entities tags is build. this should be faster than the original approach

the methods name is still up to discussion, but it should now fit a little better